src/library/librarycontrol.cpp

```diff
@@ -482,16 +482,26 @@ void LibraryControl::emitKeyEvent(QKeyEvent&& event) {
     VERIFY_OR_DEBUG_ASSERT(m_pLibraryWidget) {
         return;
     }
-    if (!m_pLibraryWidget->hasFocus() && !m_pSidebarWidget->hasFocus()) {
+    if (!QApplication::focusWindow()) {
+        qDebug()<<"Mixxx window is not focused, don't send key events";
+    }
+
+    Qt::Key key = static_cast<Qt::Key>(event.key());
+//    qDebug() << "|   key  " << key;
+
+    // If the main window has focus, any widget can receive Tab.
+    // Other keys should be sent to library widgets only to not
+    // accidentally alter spinboxes etc.
+    if (key != Qt::Key_Tab && !m_pSidebarWidget->hasFocus()
+            && !m_pLibraryWidget->getActiveView()->hasFocus()) {
         setLibraryFocus();
     }
-    auto focusWidget = QApplication::focusWidget();
-    VERIFY_OR_DEBUG_ASSERT(focusWidget) {
-        return;
+    if (key == Qt::Key_Tab && !QApplication::focusWidget()){
+        setLibraryFocus();
     }
     // Send the event pointer to the currently focused widget
     for (auto i = 0; i < event.count(); ++i) {
-        QApplication::sendEvent(focusWidget, &event);
+        QApplication::sendEvent(QApplication::focusWidget(), &event);
```

Comment:
This can be any widget .. can this become a infinite loop?

Reply:
hmm good point.
I'll restore `auto focusWidget = QApplication::focusWidget()` above.

Not an infinte loop, but multipe events could be sent to different widgets.
A widget might get disabled after the first event and consecutive events from the queue would be sent to another widget then. I don't see how that could happen from within librarycontrol but it's safer, of course.